src/app/api/gcd_result_callback/route.ts

```diff
@@ -0,0 +1,219 @@
+      signingAlgorithm,
+      keySizeBytes
+    ).toString();
+
+    // Convert signature from base64 to BigInt string
+    const signatureBigIntStr = BigInt(`0x${Buffer.from(signatureBase64, 'base64').toString('hex')}`).toString();
+
+    // Convert all values to forge.jsbn.BigInteger
+    const modulus = new forge.jsbn.BigInteger(publicKeyHex, 16);
+    const signature = new forge.jsbn.BigInteger(signatureBigIntStr, 10);
+    const encodedDigestBigInt = new forge.jsbn.BigInteger(encodedDigest, 10);
+    const exponent = new forge.jsbn.BigInteger(exponentStr, 10);
+
+    // RSA verification: signature^exponent mod modulus
+    const verified = signature.modPow(exponent, modulus);
+
+    // Compare the result with the encoded digest
+    return verified.compareTo(encodedDigestBigInt) === 0;
+
+  } catch (error) {
+    console.error('Error during RSA verification:', error);
+    return false;
+  }
+}
+
+export async function POST(request: Request) {
+  try {
+    const body = await request.json();
+
+    const { success, result, error, taskId, metadata } = body;
+
+    if (!taskId) {
+      console.error('Missing taskId in callback');
+      return Response.json(
+        { error: 'Missing taskId' },
+        { status: 400 }
+      );
+    }
+
+    if (success) {
+      console.log(chalk.magenta(`Task ${taskId} completed successfully.`));
+
+      let publicKeyBigInt = BigInt(result);
+      const publicKeyHex = publicKeyBigInt.toString(16);
+      const publicKeyBigIntjsbn = new forge.jsbn.BigInteger(publicKeyHex, 16);
+      const e = new forge.jsbn.BigInteger('010001', 16);
+      const publicKeyRaw = forge.pki.setRsaPublicKey(publicKeyBigIntjsbn, e);
+
+      const publicKeyDer = forge.asn1.toDer(forge.pki.publicKeyToAsn1(publicKeyRaw)).getBytes();
+      const publicKey = forge.util.encode64(publicKeyDer);
+
+      console.log(`${chalk.blue('🔹 Selector :')}  ${chalk.yellow(metadata.selector)}`);
+      console.log(`${chalk.blue('🔹 Domain   :')}  ${chalk.yellow(metadata.domain)}`);
+      console.log(`${chalk.blue('🔹 Public Key:')} ${chalk.green(publicKey)}`);
+
+      const isHeaderHash1SignatureValid = verifyRsaPublicKey(publicKeyHex, metadata.dkimSignature1, metadata.headerHash1, metadata.signingAlgorithm);
+      const isHeaderHash2SignatureValid = verifyRsaPublicKey(publicKeyHex, metadata.dkimSignature2, metadata.headerHash2, metadata.signingAlgorithm);
+
+      if (!isHeaderHash1SignatureValid || !isHeaderHash2SignatureValid) {
+        return Response.json(
+          {
+            error: 'Public Key is Not valid'
+          }, {
+          status: 400
+        })
+      }
+
+      await storeCalculationResult({
+        taskId,
+        result,
+        status: 'completed',
+        completedAt: new Date(),
+        metadata,
+        publicKey
+      });
+
+    } else {
+      console.error(`Task ${taskId} failed:`, error);
+
+      // Store the error in your database
+      await storeCalculationResult({
```

Comment:
The storeCalculationResult() function doesn't actually store anything on status 'failed', it just logs and returns.
(although the comment says "// Store the error in your database")
Do we want to store the error? If not, i think it would be the more intuitive to remove the `status` parameter from storeCalculationResult() and only call it on success, and on this line you can just log the error directly instead.